tests/pyfunc/test_warn_dependency_requirement_mismatches.py

```diff
@@ -14,6 +14,10 @@ def test_warn_dependency_requirement_mismatches(tmpdir):
     import cloudpickle
     import mlflow.utils.requirements_utils
 
+    class AnyStringWith(str):
+        def __eq__(self, other):
+            return self in other
```

Comment:
Can we define this class in `helper_functions.py` so we can reuse it in other tests?